### ISSUE TITLE
prevent kwarg clashes

### ISSUE THREAD

**Issue by lijok** [hynek/structlog#372]
I keep on running into this problem:
```sh
TypeError: BoundLogger._proxy_to_logger() got multiple values for argument 'method_name'
TypeError: BoundLogger._proxy_to_logger() got multiple values for argument 'method'
```
and some others, when trying to:
```py
log.error("connector method invocation failed", method=connector.formal_method_name, method_path=connector.method_path, method_name=connector.method_name)
```

I've heard from collegues of them running into the same issues with other keywords, like `event` iirc and some others

My understanding is that most keyword clashes can be prevented by changing internal method definitions to, say, prepend each kwarg with an underscore?

Let me know your thoughts, and if you think this is reasonable, I'll raise a pr to fix this

**Comment by hynek** [comment]
You're right, with your analysis, however the problem is that those method definitions aren't private at all.

`_proxy_to_logger()` is an official API for anyone implementing their own loggers and in hindsight it's a gross oversight to give the method the underscore but not to the arguments!

Event is even a bit more complicated, because it's part of the logging calls and it's syntactically impossible to pass a kw-arg called event to it.

---

Generally speaking, I would be interested to untangle all of this. But I'm afraid the only backward-compatible way would be to introduce new APIs that replace `_proxy_to_logger` and leave that one be. In fact, I wanted to get rid of all the subclassing shenanigans anyways.

**Comment by PerchunPak** [comment]
Can it at least fail silently?

Like
```py
class Logger:
    def _log(self, *args, **kwargs):
        try:
             self._real_log(*args, **kwargs)
        except Exception:
             self.exception("Internal logging error.")
```
